wasmbinding/test/custom_query_test.go

```diff
@@ -172,6 +174,97 @@ func TestQuerySpotPrice(t *testing.T) {
 	require.InEpsilonf(t, expected+swapFee, price, epsilon, fmt.Sprintf("Outside of tolerance (%f)", epsilon))
 }
 
+func TestQueryArithmeticTwap(t *testing.T) {
+	actor := RandomAccountAddress()
+	osmosis, ctx := SetupCustomApp(t, actor)
+
+	fundAccount(t, ctx, osmosis, actor, defaultFunds)
+
+	poolFunds := []sdk.Coin{
+		sdk.NewInt64Coin("uosmo", 100000000),
+		sdk.NewInt64Coin("ustar", 200000000),
+	}
+	// 20 star to 1 osmo
+	starPoolID := preparePool(t, ctx, osmosis, actor, poolFunds)
+
+	reflect := instantiateReflectContract(t, ctx, osmosis, actor)
+	require.NotEmpty(t, reflect)
+
+	// TODO: figure out why it errors when we use start time without second
```

Comment:
wdym

Reply:
Oh I see, the query errors in this case, interesting

Reply:
here's the error message: `looking for a time thats too old, not in the historical index.  Try storing the accumulator value.`

Reply:
ah the error is correct, this is a test error here =p

Committing fix here.

Reply:
Hrmm actually its not so clean, this is actually a deeper issue with the binding using Unix time for the time passing here.

The reason its not working at the start time is because its being passed in unix time in miliseocnds. However the state record has greater precision.

Trying to see more about how to fix this

Reply:
So I think what we need to do is make the TWAP binding parse from UnixMiliseconds, and then in the state machine we write at UnixMilisecond precision.

For this PR, can we change the binding to be from Unix milisecond and then call it good from that? Then in the test if we set ctx.BlockTime to be at a milisecond boundary, this should be fixed. Once we do https://github.com/osmosis-labs/osmosis/issues/2209 should also get fixed state machine side as well.